oak/server/rust/oak_runtime/src/node/mod.rs

```diff
@@ -86,63 +63,53 @@ impl std::fmt::Display for ConfigurationError {
-    let module = wasmi::Module::from_buffer(wasm_bytes)
-        .map_err(ConfigurationError::WasmiModuleInializationError)?;
-    Ok(Configuration::WasmNode {
-        module: Arc::new(module),
-    })
-}
-
-/// Checks if port is greater than 1023.
-pub fn check_port(address: &SocketAddr) -> Result<(), ConfigurationError> {
-    if address.port() > 1023 {
-        Ok(())
-    } else {
-        Err(ConfigurationError::IncorrectPort)
-    }
-}
-
-impl Configuration {
-    /// Creates a new Node instance corresponding to the [`Configuration`] `self`.
-    ///
-    /// On success returns a boxed [`Node`] that may be started by invoking the [`Node::start`]
-    /// method.
-    pub fn create_node(
-        &self,
-        config_name: &str, // Used for pretty debugging
-        runtime: RuntimeProxy,
-        entrypoint: String,
-        initial_reader: Handle,
-    ) -> Box<dyn Node> {
-        match self {
-            Configuration::LogNode => {
-                Box::new(logger::LogNode::new(config_name, runtime, initial_reader))
-            }
-            Configuration::GrpcServerNode { address } => Box::new(
-                grpc_server::GrpcServerNode::new(config_name, runtime, *address, initial_reader),
-            ),
-            Configuration::WasmNode { module } => Box::new(wasm::WasmNode::new(
-                config_name,
-                runtime,
-                module.clone(),
-                entrypoint,
-                initial_reader,
-            )),
-            Configuration::External => Box::new(external::PseudoNode::new(
-                config_name,
-                runtime,
-                initial_reader,
-            )),
-        }
+pub fn create_node(
+    application_configuration: &ApplicationConfiguration,
+    node_configuration: &NodeConfiguration,
+    runtime: RuntimeProxy,
+    initial_reader: Handle,
+) -> Result<Box<dyn Node>, ConfigurationError> {
+    let name = &node_configuration.name;
+    use oak_abi::proto::oak::application::node_configuration::ConfigType;
+    match &node_configuration.config_type {
+        Some(ConfigType::LogConfig(LogConfiguration {})) => Ok(Box::new(logger::LogNode::new(
+            name,
+            runtime,
+            initial_reader,
+        ))),
+        Some(ConfigType::GrpcServerConfig(config)) => Ok(Box::new(
+            grpc_server::GrpcServerNode::new(name, config.clone(), runtime, initial_reader)?,
+        )),
+        Some(ConfigType::WasmConfig(config)) => Ok(Box::new(wasm::WasmNode::new(
+            name,
+            application_configuration,
+            config.clone(),
+            runtime,
+            initial_reader,
+        )?)),
+        Some(ConfigType::GrpcClientConfig(_config)) => Ok(Box::new(external::PseudoNode::new(
+            name,
+            runtime,
+            initial_reader,
+        ))),
+        Some(ConfigType::RoughtimeClientConfig(_config)) => Ok(Box::new(
+            external::PseudoNode::new(name, runtime, initial_reader),
```

Comment:
Yeah, this will need to pass a copy of the node config over the Rust->C++ FFI boundary.  As you say, probably easiest to block this until `main()` flips into Rust and `oak_glue` gets reworked.